packages/shared/lib/core/wallet/actions/events-handlers/handleNewOutputEvent.ts

```diff
@@ -63,14 +63,24 @@ export async function handleNewOutputEventInternal(walletId: string, payload: Ne
         return
     }
 
+    const isAccountOutput = output.type === OutputType.Account
+
+    if (isAccountOutput) {
+        const accounts = await wallet.accounts()
+        const depositAddress = await getDepositAddress(wallet)
```

Comment:
why do we do the deposit address thing here? are we removing this line in another PR, i dont think the deposit adddress should be calculated here

Reply:
I think that when the first account output is received, we would have to update the main account id and the deposit address here, as I say in this [comment](https://github.com/iotaledger/firefly/pull/7923#discussion_r1463667808)
wdyt?